account_invoice_download_ovh/models/account_invoice_download_config.py

```diff
@@ -0,0 +1,229 @@
+                rpdf.status_code,
+            )
+        filename = "OVH_invoice_%s.pdf" % parsed_inv["invoice_number"]
+        parsed_inv["attachments"] = {filename: res}
+        return
+
+    def ovh_download(self, credentials, logs):
+        invoices = []
+        logger.info(
+            "Start to download OVH invoices with config %s and endpoint %s",
+            self.name,
+            self.ovh_endpoint,
+        )
+        try:
+            client = ovh.Client(
+                endpoint=self.ovh_endpoint,
+                application_key=self.ovh_application_key,
+                application_secret=self.ovh_application_secret,
+                consumer_key=self.ovh_consumer_key,
+            )
+        except Exception as e:
+            logs["msg"].append(
+                _(
+                    "Cannot connect to the OVH API with endpoint '%s'. "
+                    "The error message is: '%s'."
+                )
+                % (self.ovh_endpoint, str(e))
+            )
+            logs["result"] = "failure"
+            return []
+        logger.info("Starting OVH API query /me/bill (d/l config %s)", self.name)
+        params = {}
+        if self.download_start_date:
+            params = {"date.from": self.download_start_date}
+        res_ilist = client.get("/me/bill", **params)
+        logger.debug("Result of /me/bill : %s", json.dumps(res_ilist, indent=4))
+
+        for oinv_num in res_ilist:
+            logger.info(
+                "Starting OVH API query /me/bill/%s (d/l config %s)",
+                oinv_num,
+                self.name,
+            )
+            res_inv = client.get("/me/bill/%s" % oinv_num)
+            logger.debug("Result of /me/bill/%s : %s", oinv_num, json.dumps(res_inv))
+            oinv_date = res_inv["date"][:10]
+            if not res_inv["priceWithoutTax"].get("value") and not res_inv[
+                "priceWithTax"
+            ].get("value"):
+                logs["msg"].append(
+                    _("Skipping OVH invoice %s dated %s because " "the amount is 0")
+                    % (oinv_num, oinv_date)
+                )
+                continue
+            if oinv_num and oinv_num.startswith("PP_"):
+                logs["msg"].append(
+                    _(
+                        "Skipping OVH invoice %s dated %s because it is a "
+                        "special pre-paid invoice"
+                    )
+                    % (oinv_num, oinv_date)
+                )
+                continue
+
+            currency_code = res_inv["priceWithoutTax"]["currencyCode"]
+            parsed_inv = {
+                "invoice_number": oinv_num,
+                "currency": {"iso": currency_code},
+                "date": oinv_date,
+                "amount_untaxed": res_inv["priceWithoutTax"].get("value"),
+                "amount_total": res_inv["priceWithTax"].get("value"),
+            }
+            self.ovh_invoice_attach_pdf(parsed_inv, res_inv["url"])
```

Comment:
`res_inv["url"]` will give you an html representation of the invoice, the PDF is available at `res_inv["pdfUrl"]`

```suggestion
            self.ovh_invoice_attach_pdf(parsed_inv, res_inv["pdfUrl"])
```

https://api.ovh.com/console/#/me/bill/{billId}#GET